### PR TITLE
Update storage entity Account to contain just getters - Closes #2764

### LINKED ISSUE #2764: Update Account entity to have only getters and move other functions to chain specific implementation
### Expected behavior
Storage component `Account` entity should only have getters. 

### Actual behavior
Currently there are some interfaces which are changing the data also part of shared entity model.

### PR DESCRIPTION
**Pull request by diego-G**
### How to test it?
`npx mocha framework/test/mocha/unit/components/storage/entities/account.js`
`npx mocha framework/test/mocha/unit/modules/chain/components/storage/entitites/account.js`

### Review checklist

* The PR resolves #2764 
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: framework/src/modules/chain/components/storage/entities/account.js [hunk 1]
```diff
@@ -0,0 +1,503 @@
...
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const path = require('path');
+const assert = require('assert');
+const _ = require('lodash');
+const AccountEntity = require('../../../../../components/storage/entities/account');
```
Comment (nazarhussain) [inline framework/src/modules/chain/components/storage/entities/account.js]:
```suggestion
const { Account: AccountEntity } = require('../../../../../components/storage/entities');
```

File: framework/src/modules/chain/components/storage/entities/account.js [hunk 2]
```diff
@@ -0,0 +1,503 @@
...
+	increaseFieldBy: 'accounts/increase_field_by.sql',
+	decreaseFieldBy: 'accounts/decrease_field_by.sql',
+	createDependentRecord: 'accounts/create_dependent_record.sql',
+	deleteDependentRecord: 'accounts/delete_dependent_record.sql',
+	delegateBlocksRewards: 'accounts/delegate_blocks_rewards.sql',
+	syncDelegatesRank: 'accounts/sync_delegates_rank.sql',
+	insertFork: 'accounts/insert_fork.sql',
+};
+
+class Account extends AccountEntity {
```
Comment (nazarhussain) [inline framework/src/modules/chain/components/storage/entities/account.js]:
```suggestion
class ChainAccount extends AccountEntity {
```

Comment (nazarhussain) [inline framework/src/modules/chain/components/storage/entities/account.js]: That will make it more clear while searching in the code, that this Account Entity is specific extension for chain.

File: framework/src/modules/chain/init_steps/bootstrap_storage.js [hunk 3]
```diff
@@ -14,10 +14,16 @@
...
+const {
+	Account,
+	Migration,
+	Peer,
+	Round,
+} = require('../components/storage/entities');
 
 module.exports = async ({ components: { storage, logger } }, accountLimit) => {
 	try {
+		storage.registerEntity('Account', Account, { replaceExisting: true });
```
Comment (nazarhussain) [inline framework/src/modules/chain/init_steps/bootstrap_storage.js]: 👍

File: framework/src/components/storage/entities/account.js [hunk 4]
```diff
@@ -379,107 +331,6 @@ class Account extends BaseEntity {
...
 
-	/**
-	 * Create account object
-	 *
-	 * @param {Object|Array.<Object>} data
-	 * @param {Object} [_options]
-	 * @param {Object} [tx] - Transaction object
-	 * @return {*}
-	 */
-	create(data, _options, tx) {
```
Comment (lsilvs) [inline framework/src/components/storage/entities/account.js]: Would be good to keep this method but throwing `NonSupportedOperationError`

File: framework/src/components/storage/entities/account.js [hunk 5]
```diff
@@ -379,107 +331,6 @@ class Account extends BaseEntity {
...
-	/**
-	 * Update the records based on given condition
-	 *
-	 * @param {filters.Account} [filters]
-	 * @param {Object} data
-	 * @param {Object} [options]
-	 * @param {Object} tx - Transaction object
-	 * @return {*}
-	 */
-	update(filters, data, _options, tx) {
```
Comment (lsilvs) [inline framework/src/components/storage/entities/account.js]: Same as above

File: framework/src/components/storage/entities/account.js [hunk 6]
```diff
@@ -379,107 +331,6 @@ class Account extends BaseEntity {
...
-	/**
-	 * Update one record based on the condition given
-	 *
-	 * @param {filters.Account} filters
-	 * @param {Object} data
-	 * @param {Object} [options]
-	 * @param {Object} tx - Transaction object
-	 * @return {*}
-	 */
-	updateOne(filters, data, _options, tx) {
```
Comment (lsilvs) [inline framework/src/components/storage/entities/account.js]: Same as above

File: framework/src/components/storage/entities/account.js [hunk 7]
```diff
@@ -500,193 +351,6 @@ class Account extends BaseEntity {
...
 
-	/**
-	 * Delete records with following conditions
-	 *
-	 * @param {filters.Account} filters
-	 * @param {Object} [_options]
-	 * @param {Object} [tx]
-	 * @returns {Promise.<boolean, Error>}
-	 */
-	delete(filters, _options, tx = null) {
```
Comment (lsilvs) [inline framework/src/components/storage/entities/account.js]: Same as above

File: framework/src/modules/chain/components/storage/entities/account.js [hunk 8]
```diff
@@ -0,0 +1,505 @@
...
+	 * @param {BaseAdapter} adapter - Adapter to retrive the data from
+	 * @param {filters.Account} defaultFilters - Set of default filters applied on every query
+	 */
+	constructor(adapter, defaultFilters = {}) {
+		super(adapter, defaultFilters);
+
+		this.sqlDirectory = path.join(path.dirname(__filename), '../sql');
+
+		const cutomSQLs = this.loadSQLFiles('account', sqlFiles, this.sqlDirectory);
+		this.SQLs = {
```
Comment (lsilvs) [inline framework/src/modules/chain/components/storage/entities/account.js]: I like your solution but as `loadSQLFiles` returns the entire object, overwriting `this.SQLs` is not a problem.

Doing `this.SQLs = this.loadSQLFiles('account', sqlFiles, this.sqlDirectory);` should be ok.

**Pull request merged**